spec/lib/jll_property_valuation/api_spec.rb

```diff
@@ -1,951 +1,823 @@
 require "spec_helper"
 
 RSpec.describe JllPropertyValuation::Api do
-  let(:api_key) { TODO: <your token here> }
+  let(:api_key) { TODO: Your API Key here }
   let(:subject) {
     JllPropertyValuation::Api.new(
-      username: TODO: <your username here>,
-      password: TODO: <your password here>,
-      api_key: api_key
+      TODO: Username here,
+      TODO: Password here,
+      api_key
     )
   }
 
+  describe '#auth_ping' do
+    it "should return a hash with correct keys" do
+      expect(subject.auth_ping().keys).to include("Status", "Message", "Data")
+    end
+  end
+
+  describe "#auth_login" do
+    it "returns the correct keys" do
+      expect(subject.auth_login().keys).to include("Status", "Message", "Data")
+    end
+
+    it "returns the correct status" do
+      expect(subject.auth_login()["Status"]).to eq("ok")
+    end
+
+    it "returns the correct message" do
+      expect(subject.auth_login()["Message"]).to eq(nil)
+    end
+
+    it "returns the correct data" do
+      expect(subject.auth_login()["Data"].keys).to include(
+        "CompanyName",
+        "DepartmentName",
+        "Mask",
+        "TenancyId",
+        "Token",
+        "UserId",
+        "UserName"
+      )
+    end
+  end
+
   describe "#headers" do
     it "should return the correct formated headers" do
       expect(subject.send(:headers)).to eq({ "Authorization": "Bearer #{api_key}", "Accept": "application/json" })
     end
   end
 
-  describe "#get" do
-    it "should raise error if endpoint nil" do
-      expect { subject.send(:get, endpoint: nil) }.to raise_error(ArgumentError)
+  describe '#market_sale_count' do
+    let(:body) {
+      {
+        "SuburbName": "",
+        "PostcodeName": "2305",
+        "StateName": "NSW",
+        "StatisticalArea3Id": "",
+        "StatisticalArea4Id": "",
+        "GreaterCapitalCityStatisticalAreaId": "",
+        "PropertyTypeId": "hs",
+        "BedroomCount": 3,
+        "Date": "2022-01-01T00:00:00"
+      }
+    }
+
+    it "returns the correct keys" do
+      expect(subject.market_sale_count(body).keys).to include("Status", "Message", "Data")
```

Comment:
would these not be standard on any response ?  can one one factor these out into the top level request spec? 

Id say at the very least these all "behaves_like_a_jll_request" which will test status, message, body etc, and the only thing the differs are the keys in Data, it seems. We can pass those into the shared spec.